src/current/advisories/a131639.md

```diff
@@ -0,0 +1,101 @@
+
+This bug is rare in that it requires intermittent slowness of a block device backing a store, coupled with lease transfers onto the store with the intermittently slow block device, followed by async intent resolution on the slow node before it has caught up on its Raft log. Writes must touch more than one range. Single range writes, including those that benefit from the One-Phase-Commit (1PC) fast path are not affected.
+
+All CockroachDB versions prior to and including the following are impacted by this bug:
+
+- v23.1.0 - v23.1.26 and below
+- v23.2.0 - v23.2.10 and below
+- v24.1.0 - subsequent patch versions are unaffected
+
+Versions prior to 23.1 are no longer eligible for [maintenance support]({% link releases/release-support-policy.md %}), and this issue will not be addressed in those versions.
+
+## Statement
+
+This is resolved in CockroachDB by [#123442](https://github.com/cockroachdb/cockroach/pull/123442) which resolves an issue in the expiration-to-epoch lease promotion transition, where the lease's effective expiration could be allowed to regress, resulting in two nodes believing they are the leaseholder for a range.
+
+The patch has been applied to maintenance releases of CockroachDB:
+
+- [v23.1.27]({% link releases/v23.1.md%}#v23-1-27)
+- [v23.2.11]({% link releases/v23.2.md%}#v23-2-11)
+- [v24.1.1]({% link releases/v24.1.md%}#v24-1-1)
+
+This public issue is tracked by [131639](https://github.com/cockroachdb/cockroach/issues/131639).
+
+## Mitigation
+
+Users are encouraged to upgrade to [v23.1.27]({% link releases/v23.1.md%}#v23-1-27), [v23.2.11]({% link releases/v23.2.md%}#v23-2-11), [v24.1.1]({% link releases/v24.1.md%}#v24-1-1), or a later version that includes the patch.
+
+### Detection via logs
+
+While log lines alone are insufficient to confirm the occurrence of a lost write, they can be used to identify whether such a situation could have possibly occurred.
+
+First, look for log lines that indicate intermittent slowness of a block device backing one or more stores. This can be seen with log lines like:
+
+~~~
+err=heartbeat failed on epoch increment
+~~~
+
+Followed by a log line like:
+
+~~~
+incremented n.* liveness epoch to .*
+~~~
+
+Next, during the same time period as the slowness, look for delayed lease promotions on the same node as the slow disk:
+
+~~~
+applied lease after .* replication lag, client traffic may have been delayed.*epo=.*prev.*exp=.*Request.*
+~~~
+
+If all three of these log lines are present and the timestamp of the third log is less than the specified "replication lag" after the timestamp of the second log, then it is possible that a lost write occurred on the range identified in the third log line.
+
+### Detection via an audit of primary and secondary indices
```

Comment:
s/indices/indexes/
I know that "indices" is also correct, but it is not used the docs. "Indexes" is used throughout the docs instead.

Reply:
✅